File: 729-my-calendar-i/729-my-calendar-i.cpp
#define vi vector<int>
class MyCalendar {
public:
    map<vi, bool> pq;
    MyCalendar() {
    }
    
    bool book(int start, int end) {
        for(auto it =pq.begin(); it!=pq.end(); it++){
            if(start>=it->first[0] && start<it->first[1]) return false;
            if(end>it->first[0] && end<=it->first[1]) return false;
            if(start<=it->first[0] && end>=it->first[1]) return false;
        }
        pq[{start, end}]=true;
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */

File: 729-my-calendar-i/729-my-calendar-i.py
class MyCalendar:

    def __init__(self):
        self.lst = []

    def book(self, start: int, end: int) -> bool:
        if self.lst == []:
            self.lst.append([start, end])
            return True
        for lts in self.lst:
            if start < lts[1] and end > lts[0]:
                return False
        self.lst.append([start, end])
        return True


# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)